### PR TITLE
MODFEE-152 Add fee/fine owner column to refund report, change paid/transferred columns logic

### PR DESCRIPTION
**Pull request by Illia-Dolzhenko**
Resolves: [MODFEE-152](https://issues.folio.org/browse/MODFEE-152)

### CODE REVIEW

File: src/main/java/org/folio/rest/service/report/RefundReportService.java [hunk 1]
```diff
@@ -231,20 +233,29 @@ else if (actionIsOfType(feeFineAction, REFUND)) {
...
       }
 
       if (accountCtx != null) {
+        if(isRefundedToPatron(feeFineAction)){
+          reportEntry.withPaidAmount(accountCtx.paidAmount.toString());
+        }else if(isRefundedToBursar(feeFineAction)){
+          reportEntry.withTransferredAmount(accountCtx.transferredAmount.toString())
+            .withTransferAccount(singleOrDefaultMessage(accountCtx.transferAccounts,
+            MULTIPLE_MESSAGE));
+        }else {
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:249]:
```suggestion
        } else {
```

File: src/main/java/org/folio/rest/service/report/RefundReportService.java [hunk 2]
```diff
@@ -231,20 +233,29 @@ else if (actionIsOfType(feeFineAction, REFUND)) {
...
+          .withDateBilled(formatDate(account.getMetadata().getCreatedDate(), ctx.timeZone))
+          .withFeeFineOwner(account.getFeeFineOwner());
       } else {
         log.error("Refund report - account is null, refund action {}", feeFineAction.getId());
       }
 
       if (accountCtx != null) {
+        if(isRefundedToPatron(feeFineAction)){
+          reportEntry.withPaidAmount(accountCtx.paidAmount.toString());
+        }else if(isRefundedToBursar(feeFineAction)){
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:245]:
```suggestion
        } else if(isRefundedToBursar(feeFineAction)){
```

File: src/main/java/org/folio/rest/service/report/RefundReportService.java [hunk 3]
```diff
@@ -231,20 +233,29 @@ else if (actionIsOfType(feeFineAction, REFUND)) {
...
           .withBilledAmount(formatMonetaryValue(account.getAmount()))
-          .withDateBilled(formatDate(account.getMetadata().getCreatedDate(), ctx.timeZone));
+          .withDateBilled(formatDate(account.getMetadata().getCreatedDate(), ctx.timeZone))
+          .withFeeFineOwner(account.getFeeFineOwner());
       } else {
         log.error("Refund report - account is null, refund action {}", feeFineAction.getId());
       }
 
       if (accountCtx != null) {
+        if(isRefundedToPatron(feeFineAction)){
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:243]:
```suggestion
        if (isRefundedToPatron(feeFineAction)){
```

File: src/main/java/org/folio/rest/service/report/RefundReportService.java [hunk 4]
```diff
@@ -64,6 +64,8 @@
   private static final int REPORT_ROWS_LIMIT = 1_000_000;
   private static final String MULTIPLE_MESSAGE = "Multiple";
   private static final String SEE_FEE_FINE_DETAILS_PAGE_MESSAGE = "See Fee/fine details page";
+  private static final String REFUNDED_TO_PATRON = "Refunded to patron";
+  private static final String REFUNDED_TO_BURSAR = "Refunded to Bursar";
   private static final  LocaleSettings FALLBACK_LOCALE_SETTINGS =
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:69]:
```suggestion
  private static final LocaleSettings FALLBACK_LOCALE_SETTINGS =
```

File: src/main/java/org/folio/rest/service/report/RefundReportService.java [hunk 5]
```diff
@@ -231,20 +233,29 @@ else if (actionIsOfType(feeFineAction, REFUND)) {
...
+          .withDateBilled(formatDate(account.getMetadata().getCreatedDate(), ctx.timeZone))
+          .withFeeFineOwner(account.getFeeFineOwner());
       } else {
         log.error("Refund report - account is null, refund action {}", feeFineAction.getId());
       }
 
       if (accountCtx != null) {
+        if (isRefundedToPatron(feeFineAction)){
+          reportEntry.withPaidAmount(accountCtx.paidAmount.toString());
+        } else if (isRefundedToBursar(feeFineAction)){
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:245]:
```suggestion
        } else if (isRefundedToBursar(feeFineAction)) {
```

Comment (roman-barannyk) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:245]:
```suggestion
        } else if (isRefundedToBursar(feeFineAction)) {
```

File: src/main/java/org/folio/rest/service/report/RefundReportService.java [hunk 6]
```diff
@@ -231,20 +233,29 @@ else if (actionIsOfType(feeFineAction, REFUND)) {
...
+          reportEntry.withTransferredAmount(accountCtx.transferredAmount.toString())
+            .withTransferAccount(singleOrDefaultMessage(accountCtx.transferAccounts,
+            MULTIPLE_MESSAGE));
+        } else {
+          reportEntry.withPaidAmount(accountCtx.paidAmount.toString())
+            .withTransferredAmount(accountCtx.transferredAmount.toString())
+            .withTransferAccount(singleOrDefaultMessage(accountCtx.transferAccounts,
+              MULTIPLE_MESSAGE));;
```
Comment (roman-barannyk) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:253]:
```suggestion
              MULTIPLE_MESSAGE));
```

File: src/main/java/org/folio/rest/service/report/RefundReportService.java [hunk 7]
```diff
@@ -231,20 +233,29 @@ else if (actionIsOfType(feeFineAction, REFUND)) {
...
+          reportEntry.withPaidAmount(accountCtx.paidAmount.toString());
+        } else if (isRefundedToBursar(feeFineAction)) {
+          reportEntry.withTransferredAmount(accountCtx.transferredAmount.toString())
+            .withTransferAccount(singleOrDefaultMessage(accountCtx.transferAccounts,
+            MULTIPLE_MESSAGE));
+        } else {
+          reportEntry.withPaidAmount(accountCtx.paidAmount.toString())
+            .withTransferredAmount(accountCtx.transferredAmount.toString())
+            .withTransferAccount(singleOrDefaultMessage(accountCtx.transferAccounts,
+              MULTIPLE_MESSAGE));
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/service/report/RefundReportService.java:253]: I think we can avoid duplicating code here if for our conditions above we used something like `isRefundedToPatron && ! isRefundedToBursar` and `isRefundedToBursar && ! isRefundedToPatron `

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**